components/feature/downloads/src/test/java/mozilla/components/feature/downloads/DownloadManagerTest.kt

```diff
@@ -99,6 +102,25 @@ class DownloadManagerTest {
         assertFalse(downloadCompleted)
     }
 
+    @Test
+    fun `no null or empty headers can be added to the DownloadManager`() {
+        val request = Request(Uri.parse(download.url))
+
+        var wasAdded = request.addRequestHeaderSafely("User-Agent", "")
+
+        assertFalse(wasAdded)
+
+        wasAdded = request.addRequestHeaderSafely("User-Agent", null)
+
+        assertFalse(wasAdded)
+
+        val fireFox = "Mozilla/5.0 (Windows NT 5.1; rv:7.0.1) Gecko/20100101 Firefox/7.0.1"
+
+        wasAdded = request.addRequestHeaderSafely("User-Agent", fireFox)
+
+        assertTrue(wasAdded)
+    }
```

Comment:
Instead of using the `Boolean` which seems to only exist for the test. You could call this on a mock request and validate that `addRequestHeader()` was called or not.